build_runner/lib/src/entrypoint/run.dart

```diff
@@ -10,7 +10,7 @@ import 'options.dart';
 
 /// A common entrypoint to parse command line arguments and build or serve with
 /// [builders].
-Future<int> run(List<String> args, List<BuilderApplication> builders) {
+Future<int> run(List<String> args, List<BuilderApplication> builders) async {
   var runner = new BuildCommandRunner(builders);
-  return runner.run(args);
+  return (await runner.run(args)) ?? 0;
```

Comment:
Might could be nice to add a note here that `--help` is the only command we expect to return `null`.

Reply:
Anything that returns null and doesn't `throw` should be fine to use `0`.  I don't think we need to add a comment based on details that could change outside this code